node/src/components/consensus/era_supervisor.rs

```diff
@@ -72,6 +72,9 @@ const BONDED_ERAS: u64 = DEFAULT_UNBONDING_DELAY - AUCTION_DELAY;
 pub struct EraSupervisor<I> {
     /// A map of active consensus protocols.
     /// A value is a trait so that we can run different consensus protocol instances per era.
+    ///
+    /// This map always contains exactly `2 * BONDED_ERAS + 1` entries, with the last one being the
```

Comment:
Oh, how about we make it a `[EraId<I>, 2 * BONDED_ERAS + 1]` ? Although, I think `BONDED_ERAS` won't be a constant any more after Karan's work.

Reply:
Yes, I was about to do that as well, but then hesitated.
At least I'd do it in a separate PR.
We could make it a `VecDeque` or something.

Reply:
Look [here](https://github.com/CasperLabs/casper-node/pull/403/files#diff-e22d39c8a2209dfe9a4cc873a3ce8d2635b7425263098d8df3539948108ffefeR79). No point in changing it now since #403 will void it.